src/app/Annotation.tsx

```diff
@@ -141,115 +266,20 @@ export const useCreationBounds = () => {
-          nodeRef={ref}
-          css={{ cursor: "inherit", ...css }}
-        />
-      );
-    }
-    case "SELECT": {
-      const isSelected = Boolean(selectedAnnotations[props.id]);
-      // When multiple selections are made, we want to show action menu on
-      // the annotation which was selected first from the set.
-      const isFirstSelection = Object.keys(selectedAnnotations)[0] === props.id;
-      return (
-        <Rnd
-          allowAnyClick
-          style={{
-            position: "absolute",
-            border: isSelected ? "3px solid black" : props.border,
-            backgroundColor: props.backgroundColor,
-          }}
-          position={{
-            x: props.left + MYSTERIOUS_RND_OFFSET,
-            y: props.top + MYSTERIOUS_RND_OFFSET,
-          }}
-          size={{ height: props.height, width: props.width }}
-          onClick={(e: React.MouseEvent<HTMLElement>) => {
-            e.stopPropagation();
-            const shiftNotPressed = !e.shiftKey;
-            if (shiftNotPressed) {
-              dispatch({ type: "DESELECT_ALL_ANNOTATION" });
-            }
-            if (isSelected) {
-              dispatch({ type: "DESELECT_ANNOTATION", payload: props.id });
-            } else {
-              dispatch({ type: "SELECT_ANNOTATION", payload: props.id });
-            }
-          }}
-          css={{
-            ...css,
-            border: isSelected ? "2px solid black" : css.border,
-          }}
-          onDragStop={(_, delta) => {
-            dispatch({
-              type: "MOVE_ANNOTATION",
-              payload: {
-                id: props.id,
-                x: delta.x - MYSTERIOUS_RND_OFFSET,
-                y: delta.y - MYSTERIOUS_RND_OFFSET,
-              },
-            });
-          }}
-          onResize={(_, __, ref, ___, el) => {
-            dispatch({
-              type: "RESIZE_ANNOTATION",
-              payload: {
-                id: props.id,
-                width: ref.offsetWidth,
-                height: ref.offsetHeight,
-                x: el.x - MYSTERIOUS_RND_OFFSET,
-                y: el.y - MYSTERIOUS_RND_OFFSET,
-              },
-            });
-          }}>
-          {props.type}
-          {isFirstSelection && (
-            <FieldLayerActionMenu
-              onDelete={() => {
-                dispatch({
-                  type: "DELETE_ANNOTATION",
-                  payload: Object.keys(selectedAnnotations),
-                });
-              }}
-              onFieldTypeChange={(value) => {
-                dispatch({
-                  type: "SET_ANNOTATION_TYPE",
-                  payload: {
-                    ids: Object.keys(selectedAnnotations),
-                    type: value,
-                  },
-                });
-              }}
-            />
-          )}
-        </Rnd>
-      );
-    }
+  const dispatch = useDispatch();
+  switch (step) {
+    case 0:
+      return FieldLayerTools(props, ref, state, dispatch);
```

Comment:
Why are these function calls instead of a component? Could we change the API to:

```
<FieldLayerTools props={props} ref={ref} />
```

Inside the component, we'll be able to use `useSelector` and `useDispatch` to get at `state` and `dispatch`, so we won't need to pass those as props anymore.

Reply:
This is so stupid of me. I just wanted to move the component out to layer file and did not even think about this basic thing once. Amazing suggestion 👍